crates/router/src/core/payments/transformers.rs

```diff
@@ -269,6 +270,7 @@ pub fn payments_to_payments_response<R, Op>(
     operation: &Op,
     ephemeral_key_option: Option<ephemeral_key::EphemeralKey>,
     session_tokens: Vec<api::SessionToken>,
+    mandate_data: Option<api_models::payments::MandateData>,
```

Comment:
add `[auth_based]` attribute to this. This field should only be sent in the response if `api_key` is used. Check how it is done for `payment_method_data` in `PaymentsResponse`.

Reply:
Okay